release/release.go

```diff
@@ -622,6 +679,40 @@ func LatestRC(ctx context.Context, owner, repo, k8sVersion string, client *githu
 
 }
 
+// rke2ChartVersion will return the version of the rke2 chart from the chart versions file
+func rke2ChartsVersion(branchVersion string) (map[string]chart, error) {
+
+	chartsMap := make(map[string]chart)
+	c := charts{}
+	chartVersionsURL := "https://raw.githubusercontent.com/rancher/rke2/" + branchVersion + "/charts/" + rke2ChartsVersionsFile
+	resp, err := http.Get(chartVersionsURL)
+	if err != nil {
+		logrus.Debugf("failed to fetch url %s: %v", chartVersionsURL, err)
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		logrus.Debugf("status error: %v when fetching %s", resp.StatusCode, err)
+		return nil, err
+	}
+
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		logrus.Debugf("read body error: %v", err)
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if err := yaml.Unmarshal(b, &c); err != nil {
+		return nil, err
+	}
+	for _, chart := range c.Charts {
```

Comment:
Can you move the chartsData map down to here and use the length of c.Charts in the make call?